In [35]:
from IPython.display import Image
img = Image(url="https://upload.wikimedia.org/wikipedia/commons/thumb/e/ed/Pandas_logo.svg/1280px-Pandas_logo.svg.png", width=250)
img

- ### 셀이 선택되면 **M**은 `MarkDown`, **Y**는 `Python Code`

# Pandas

## 개요

**관계형** 또는 **레이블이 된** 데이터로 쉽고 직관적 으로 작업할 수 있도록 설계되었고, 빠르고, 유연한 데이터 구조를 제공하는 Python 패키지입니다.

또한, 어떤 언어로도 사용할 수 있는 가장 **강력하고 유연한 오픈 소스 데이터 분석 / 조직 도구**입니다.

Pandas는 다음의 종류의 데이터에 **적합한 분석 패키지**입니다.

- SQL 테이블 또는 Excel 스프레드 시트에서와 같은 열과 행으로 이루어진 테이블 형식 데이터
- 정렬되고 정렬되지 않은 시계열 데이터
- 다른 형태의 관찰 / 통계 데이터 세트

## Pandas의 Alias 지정

pandas는 `pd`의 alias를 사용합니다.

In [36]:
import pandas

In [37]:
pandas

<module 'pandas' from 'c:\\Users\\jshkr\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\site-packages\\pandas\\__init__.py'>

In [38]:
import pandas as pd

In [39]:
pd

<module 'pandas' from 'c:\\Users\\jshkr\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\site-packages\\pandas\\__init__.py'>

In [40]:
pd.__version__

'2.2.2'

## Series

Pandas의 Series는 1차원 배열로서 다음의 특징을 가집니다.

- 데이터를 담는 **차원 배열 구조**를 가집니다.
- **인덱스(index)를 사용 가능**합니다.
- **데이터 타입**을 가집니다. (dtype)

### Series의 생성

numpy array 로 생성한 경우

In [41]:
# numpy 생성
import numpy as np
arr = np.arange(100, 105)
arr

array([100, 101, 102, 103, 104])

In [42]:
s = pd.Series(arr)
s

0    100
1    101
2    102
3    103
4    104
dtype: int32

`dtype`을 지정하여 생성한 경우

In [43]:
s = pd.Series(arr, dtype='int32')
s

0    100
1    101
2    102
3    103
4    104
dtype: int32

##### list로 생성한 경우

In [44]:
a = pd.Series(['부장', '차장', '대리', '사원', '인턴'])
# a == print(a)
print(a)

0    부장
1    차장
2    대리
3    사원
4    인턴
dtype: object


##### 다양한 타입(type)의 데이터를 섞은 경우

Series에 다양한 데이터 타입의 데이터로 생성 시, **object** 타입으로 생성됩니다.

In [45]:
a = pd.Series([91, 2.5, '스포츠', 4, 5.16])
a

0      91
1     2.5
2     스포츠
3       4
4    5.16
dtype: object

### 연습문제

다음의 Series를 생성하세요
- dtype은 `float32`가 되도록 생성하세요

In [46]:
# 코드를 입력해주세요
a = pd.Series([3, 5, 7, 9, 11], dtype='float32')
a

0     3.0
1     5.0
2     7.0
3     9.0
4    11.0
dtype: float32

In [47]:
_list = [3,5,7,9,11]
pd.Series(_list, dtype='float32')

0     3.0
1     5.0
2     7.0
3     9.0
4    11.0
dtype: float32

<p><strong>[출력 결과]</strong></p>
<pre>
0     3.0
1     5.0
2     7.0
3     9.0
4    11.0
dtype: float32</pre>


In [48]:
# 코드를 입력해 주세요
pd.Series(['가', '나','다','라','마'])

0    가
1    나
2    다
3    라
4    마
dtype: object

In [49]:
# list 로 만드는 방법
_list = list('가나다라마')
pd.Series(_list)

0    가
1    나
2    다
3    라
4    마
dtype: object

<p><strong>[출력 결과]</strong></p>
<pre>
0   가
1   나
2   다
3   라
4   마
dtype: object</pre>

### 인덱싱 (indexing)

In [50]:
s = pd.Series(['부장', '차장', '대리', '사원', '인턴'])
s

0    부장
1    차장
2    대리
3    사원
4    인턴
dtype: object

Series 생성 시 0부터 순차적으로 부여되는 index를 확인할 수 있습니다.

이를 `RangeIndex` 라고 부릅니다.

In [51]:
s.index

RangeIndex(start=0, stop=5, step=1)

인덱싱 사례

In [52]:
s[0]

'부장'

### Fancy Indexing

In [53]:
idx = [1, 3]
s[idx]

1    차장
3    사원
dtype: object

In [54]:
np.arange(1, 4, 2)

array([1, 3])

In [55]:
idx = np.arange(1, 4, 2)
s[idx]

1    차장
3    사원
dtype: object

### boolean indexing

조건식을 만들어서 특정 조건에 대하여 `True`에 해당하는 값만 필터링 할 수 있습니다.

In [56]:
np.random.seed(0)
s = pd.Series(np.random.randint(10000, 20000, size=(10,)))
s

0    12732
1    19845
2    13264
3    14859
4    19225
5    17891
6    14373
7    15874
8    16744
9    13468
dtype: int32

boolean series를 생성 후 index로 활용하여 필터합니다.

In [57]:
s > 15000

0    False
1     True
2    False
3    False
4     True
5     True
6    False
7     True
8     True
9    False
dtype: bool

In [58]:
# 15000 이상인 데이터 필터
s[s > 15000]

1    19845
4    19225
5    17891
7    15874
8    16744
dtype: int32

In [59]:
idx = (s > 15000) & (s < 18000)
s[idx]

5    17891
7    15874
8    16744
dtype: int32

기본 값으로 부여되는 `RangeIndex` 에 사용자 정의의 `index`를 **지정**할 수 있습니다

In [60]:
s = pd.Series(['마케팅', '경영', '개발', '기획', '인사'], index=['a', 'b', 'c', 'd', 'e'])
s

a    마케팅
b     경영
c     개발
d     기획
e     인사
dtype: object

In [61]:
s.index

Index(['a', 'b', 'c', 'd', 'e'], dtype='object')

사용자 정의의 index 부여 시 **변경된 index**로 조회 가능합니다.

In [62]:
s['c']

'개발'

In [63]:
s[['a', 'd']]

a    마케팅
d     기획
dtype: object

먼저, `Series`를 생성 후 index 속성 값에 새로운 index를 할당하여 인덱스를 지정할 수 있습니다.

In [64]:
s = pd.Series(['마케팅', '경영', '개발', '기획', '인사'])
s.index

RangeIndex(start=0, stop=5, step=1)

In [65]:
list('abcde')

['a', 'b', 'c', 'd', 'e']

In [66]:
s.index = list('abcde')
s

a    마케팅
b     경영
c     개발
d     기획
e     인사
dtype: object

### 연습문제

다음의 Series를 생성하고 `sample` 변수에 대입하고 출력하세요

In [67]:
# 코드를 입력해 주세요
#_list = list('가나다라마')
list_number = [10, 20, 30, 40, 50]
# sample = pd.Series(list_number, index=_list)
# print(sample)
# print('\n')

# 다른 방법 s.index

sample = pd.Series(list_number)
sample.index = _list
print(sample)

가    10
나    20
다    30
라    40
마    50
dtype: int64


<p><strong>[출력 결과]</strong></p>
<pre>
가  10
나  20
다  30
라  40
마  50
dtype: int64</pre>

`sample` 중 **'나' 와 '라'** 데이터를 조회하세요

In [68]:
# 코드를 입력하세요
idx = ['나','라']
sample[idx]

나    20
라    40
dtype: int64

<p><strong>[출력 결과]</p></strong>
<pre>
나  20
라  40
dtype: int64</pre>

In [69]:
np.random.seed(20)
sample2 = pd.Series(np.random.randint(100, 200, size=(15,)))
sample2

0     199
1     190
2     115
3     195
4     128
5     190
6     109
7     120
8     175
9     122
10    171
11    134
12    196
13    140
14    185
dtype: int32

`sample2` 중 160 이하인 데이터만 필터하세요

In [70]:
# 코드를 입력해 주세요
sample2[sample2 <= 160]

2     115
4     128
6     109
7     120
9     122
11    134
13    140
dtype: int32

<p><strong>[출력 결과]</strong></p>
<pre>2     115
4     128
6     109
7     120
9     122
11    134
13    140
dtype: int64</pre>

`sample2` 중 **130 이상 170 이하**인 데이터만 필터하세요

In [71]:
# 코드를 입력해주세요
idx = (sample2 >= 130) & (sample2 <= 170)
sample2[idx]

11    134
13    140
dtype: int32

<p><strong>[출력 결과]</p></strong>
<pre>
11  134
13  140
dtype: int64</pre>

## 속성 (attribute)

### values

`values` 는 Series 데이터 값(value)만 **numpy array** 형식으로 가져 옵니다.

In [72]:
s

a    마케팅
b     경영
c     개발
d     기획
e     인사
dtype: object

In [73]:
s.values

array(['마케팅', '경영', '개발', '기획', '인사'], dtype=object)

### ndim - 차원

`Series`는 1차원 자료구조이기 때문에 `ndim` 출력시 **1**이 출력됩니다.

In [74]:
s.ndim

1

### shape

shape은 데이터의 모양(shape)을 알아보기 위하여 사용하는데, `Series`의 shape은 **데이터의 개수**를 나타냅니다.


**튜플(tuple)** 형식으로 출력됩니다.

In [75]:
s.shape

(5,)

### NaN (Not a Number)

Pandas에서 **NaN 값**은 비어있는 **결측치 데이터**를 의미합니다.

임의로 비어있는 값을 대입하고자 할 때는 **numpy의 nan (np.nan)** 을 입력합니다.


In [76]:
s = pd.Series([
    '선화', '강호', np.nan, '소정', '우영'
])
print(s)

0     선화
1     강호
2    NaN
3     소정
4     우영
dtype: object


### 연습문제

다음과 같은 `Series` 를 생성해 주세요

In [77]:
# 코드를 입력해 주세요
_list = list('가나다라마')
a = pd.Series(['apple', np.nan, 'banana', 'kiwi', 'gubong'])
a.index = _list
a

가     apple
나       NaN
다    banana
라      kiwi
마    gubong
dtype: object

<p><strong>[출력 결과]</strong></p>
<pre>가     apple
나       NaN
다    banana
라      kiwi
마    gubong
dtype: object</pre>

### 결측치 (NaN) 값 처리

`isnull()` 과 `isna()`은 **NaN** 값을 찾는 함수입니다.

`isnull()` 과 `isna()`는 결과가 동일합니다

In [78]:
s

0     선화
1     강호
2    NaN
3     소정
4     우영
dtype: object

In [79]:
s.isnull()

0    False
1    False
2     True
3    False
4    False
dtype: bool

In [80]:
s.isna()

0    False
1    False
2     True
3    False
4    False
dtype: bool

In [81]:
s[s.isnull()]

2    NaN
dtype: object

In [82]:
s[s.isna()]

2    NaN
dtype: object

`notnull()`은 **NaN**값이 아닌, 즉 비어있지 않은 데이터를 찾는 함수입니다.

In [83]:
s.notnull()

0     True
1     True
2    False
3     True
4     True
dtype: bool

In [84]:
s.notna()

0     True
1     True
2    False
3     True
4     True
dtype: bool

In [85]:
s[s.notnull()]

0    선화
1    강호
3    소정
4    우영
dtype: object

### 연습문제

In [86]:
sample = pd.Series(['IT서비스', np.nan, '반도체', np.nan, '바이오', '자율주행'])
sample

0    IT서비스
1      NaN
2      반도체
3      NaN
4      바이오
5     자율주행
dtype: object

`sample` 중 결측치 데이터만 필터하세요

In [87]:
# 코드를 입력하세요
sample[sample.isnull()]

1    NaN
3    NaN
dtype: object

<p><strong>[출력 결과]</p></strong>
<pre>
1   NaN
3   NaN
dytpe: object

`sample` 중 결측치가 아닌 데이터만 필터하세요

In [88]:
# 코드를 입력해 주세요
sample[sample.notna()]

0    IT서비스
2      반도체
4      바이오
5     자율주행
dtype: object

<p><strong>[출력 결과]</strong></p>
<pre>0    IT서비스
2      반도체
4      바이오
5     자율주행
dtype: object</pre>

## 슬라이싱

**(주의)** 숫자형 index로 접근할 때는 뒷 index가 포함되지 않습니다

In [89]:
s = pd.Series(np.arange(100, 150, 10))
s

0    100
1    110
2    120
3    130
4    140
dtype: int32

In [90]:
s[1:3]

1    110
2    120
dtype: int32

In [91]:
s[1:]

1    110
2    120
3    130
4    140
dtype: int32

In [92]:
s[::2]

0    100
2    120
4    140
dtype: int32

새롭게 지정한 인덱스(문자열)는 시작 index와 끝 index **모두 포함**합니다.

In [93]:
s.index = list('가나다라마')
s

가    100
나    110
다    120
라    130
마    140
dtype: int32

In [94]:
s1 = pd.Series(np.arange(100, 250, 10))
s1

0     100
1     110
2     120
3     130
4     140
5     150
6     160
7     170
8     180
9     190
10    200
11    210
12    220
13    230
14    240
dtype: int32

In [95]:
s1[1::2]

1     110
3     130
5     150
7     170
9     190
11    210
13    230
dtype: int32

In [96]:
s1[::2]

0     100
2     120
4     140
6     160
8     180
10    200
12    220
14    240
dtype: int32

In [97]:
s1[1::3]

1     110
4     140
7     170
10    200
13    230
dtype: int32

In [98]:
s['나':'라']

나    110
다    120
라    130
dtype: int32

### 연습문제

In [99]:
np.random.seed(0)
sample = pd.Series(np.random.randint(100, 200, size=(10,)))
sample

0    144
1    147
2    164
3    167
4    167
5    109
6    183
7    121
8    136
9    187
dtype: int32

`sample`에서 다음과 같은 결과를 가지도록 **슬라이싱** 하세요

In [100]:
# 코드를 입력해 주세요
sample[2:7]

2    164
3    167
4    167
5    109
6    183
dtype: int32

<p><strong>[출력 결과]</p></strong>
<pre>
2   164
3   167
4   167
5   109
6   183
dtype: int64</pre>

In [101]:
np.random.seed(0)
sample2 = pd.Series(np.random.randint(100, 200, size=(10,)), index=list('가나다라마바사아자차'))
sample2

가    144
나    147
다    164
라    167
마    167
바    109
사    183
아    121
자    136
차    187
dtype: int32

`sample2`에서 다음과 같은 결과를 가지도록 **슬라이싱**하세요

In [102]:
# 코드를 입력해 주세요
sample2['바':'차']

바    109
사    183
아    121
자    136
차    187
dtype: int32

<p><strong>[출력 결과]</strong></p>
<pre>바    109
사    183
아    121
자    136
차    187
dtype: int64</pre>

In [103]:
# 코드를 입력해 주세요
sample2['가':'다']

가    144
나    147
다    164
dtype: int32

<p><strong>[출력 결과]</strong></p>
<pre>가    144
나    147
다    164
dtype: int64</pre>

In [104]:
# 코드를 입력해 주세요
sample2['나':'바']

나    147
다    164
라    167
마    167
바    109
dtype: int32

<p><strong>[출력 결과]</strong></p>
<pre>나    147
다    164
라    167
마    167
바    109
dtype: int64</pre>

### DataFrame

[도큐먼트](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.html)

`pd.DataFrame`

- 2차원 데이터 구조 (Excel 데이터 시트를 생각하시면 됩니다)
- 행(row), 열(column)으로 구성되어 있습니다.
- 각 열(column)은 각각의 데이터 타입 (dtype)을 가집니다.

### 생성

**list 를 통한 생성**할 수 있습니다. DataFrame을 만들 때는 **2차원 list를 대입**합니다.

In [105]:
pd.DataFrame([[1, 2, 3], 
              [4, 5, 6], 
              [7, 8, 9]])

0  1  2
0  1  2  3
1  4  5  6
2  7  8  9

아래 예제와 같이 **columns를 지정**하면, DataFrame의 각 열에 대한 컬럼명이 붙습니다.

In [106]:
pd.DataFrame([[1, 2, 3], 
              [4, 5, 6], 
              [7, 8, 9]], columns=['가', '나', '다'])

가  나  다
0  1  2  3
1  4  5  6
2  7  8  9

**dictionary를 통한 생성**도 가능합니다.

편리한 점은 dictionary의 **key 값이 자동으로 column 명으로 지정**됩니다.

In [107]:
data = {
    'name': ['Kim', 'Lee', 'Park'], 
    'age': [24, 27, 34], 
    'children': [2, 1, 3]
}

In [108]:
pd.DataFrame(data, index = [1,2,3])

name  age  children
1   Kim   24         2
2   Lee   27         1
3  Park   34         3

### 속성

DataFrame은 다음의 **속성**을 가집니다.

- **index**: index (기본 값으로 RangeIndex)
- **columns**: column 명
- **values**: numpy array형식의 데이터 값
- **dtypes**: column 별 데이터 타입
- **T**: DataFrame을 전치(Transpose)

In [109]:
data = {
    'name': ['Kim', 'Lee', 'Park'], 
    'age': [24, 27, 34], 
    'children': [2, 1, 3]
}

df = pd.DataFrame(data)
df

name  age  children
0   Kim   24         2
1   Lee   27         1
2  Park   34         3

In [110]:
df.index

RangeIndex(start=0, stop=3, step=1)

In [111]:
df.columns

Index(['name', 'age', 'children'], dtype='object')

In [112]:
df.values

array([['Kim', 24, 2],
       ['Lee', 27, 1],
       ['Park', 34, 3]], dtype=object)

데이터프레임에서의 `dtypes` 속성은 칼럼별 `dtype`을 출력합니다

In [113]:
df.dtypes

name        object
age          int64
children     int64
dtype: object

`df.T`는 `DataFrame`을 **Transpose** 시켜줍니다.

In [114]:
df.index=[1,2,3]
df.T

1    2     3
name      Kim  Lee  Park
age        24   27    34
children    2    1     3

### index 지정

In [115]:
df

name  age  children
1   Kim   24         2
2   Lee   27         1
3  Park   34         3

In [116]:
df.index = list('abc')
df

name  age  children
a   Kim   24         2
b   Lee   27         1
c  Park   34         3

(참고) DataFrame의 indexing / slcing은 나중에 세부적으로 다루도록 하겠습니다.

### column 다루기

DataFrame에 key 값으로 column의 이름을 지정하여 column을 선택할 수 있습니다.

1개의 column을 가져올 수 있으며, **1개의 cloumn 선택시 Series**가 됩니다.

In [117]:
df['name']

a     Kim
b     Lee
c    Park
Name: name, dtype: object

In [118]:
type(df['name'])

pandas.core.series.Series

2개 이상의 column 선택은 **fancy indexing으로 가능**합니다.

In [119]:
df[['name', 'children']]

name  children
a   Kim         2
b   Lee         1
c  Park         3

(참고) column에 대한 slicing도 가능하지만 이 부분도 나중에 다루도록 하겠습니다.

**rename**으로 column명 변경 가능합니다.

DataFrame.rename(columns={**'바꾸고자 하는 column명'** : **'바꿀 컬럼명'**})

In [120]:
df.rename(columns={'name': '이름'})

이름  age  children
a   Kim   24         2
b   Lee   27         1
c  Park   34         3

In [121]:
df.rename(columns={'age': '나이'})

name  나이  children
a   Kim  24         2
b   Lee  27         1
c  Park  34         3

`inplace=True` 옵션으로 변경사항을 바로 적용할 수 있습니다.

In [122]:
df.rename(columns={'name': '이름'}, inplace=True)
df

이름  age  children
a   Kim   24         2
b   Lee   27         1
c  Park   34         3

### 연습문제

다음의 DataFrame을 생성하세요

- 생성된 DataFrame은 df 변수에 할당합니다.

In [123]:
# 코드를 입력해 주세요
df = pd.DataFrame({
    'food' : ['KFC', 'McDonald','SchoolFood'],
    'price' : [1000, 2000, 2500],
    'rating' : [4.5, 3.9, 4.2]
})
df

food  price  rating
0         KFC   1000     4.5
1    McDonald   2000     3.9
2  SchoolFood   2500     4.2

<p><strong>[출력 결과]</strong></p>
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>food</th>
      <th>price</th>
      <th>rating</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>KFC</td>
      <td>1000</td>
      <td>4.5</td>
    </tr>
    <tr>
      <th>1</th>
      <td>McDonald</td>
      <td>2000</td>
      <td>3.9</td>
    </tr>
    <tr>
      <th>2</th>
      <td>SchoolFood</td>
      <td>2500</td>
      <td>4.2</td>
    </tr>
  </tbody>
</table>
</div>

food column과 rating column만 선택하여 출력하세요

In [124]:
# 코드를 입력해 주세요
df[['food', 'rating']]

food  rating
0         KFC     4.5
1    McDonald     3.9
2  SchoolFood     4.2

<p><strong>[출력 결과]</strong></p>
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>food</th>
      <th>rating</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>KFC</td>
      <td>4.5</td>
    </tr>
    <tr>
      <th>1</th>
      <td>McDonald</td>
      <td>3.9</td>
    </tr>
    <tr>
      <th>2</th>
      <td>SchoolFood</td>
      <td>4.2</td>
    </tr>
  </tbody>
</table>
</div>

In [125]:
type(df[['food', 'rating']])

pandas.core.frame.DataFrame

food column명을 place로 변경해 주세요

In [126]:
# 코드를 입력해 주세요
df.rename(columns={'food': 'place'}, inplace=True)

<p><strong>[출력 결과]</strong></p>
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>place</th>
      <th>price</th>
      <th>rating</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>KFC</td>
      <td>1000</td>
      <td>4.5</td>
    </tr>
    <tr>
      <th>1</th>
      <td>McDonald</td>
      <td>2000</td>
      <td>3.9</td>
    </tr>
    <tr>
      <th>2</th>
      <td>SchoolFood</td>
      <td>2500</td>
      <td>4.2</td>
    </tr>
  </tbody>
</table>
</div>

## 제출

제출을 위해 아래와 같은 DataFrame을 생성하여 `result_df`에 저장하세요.

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>name</th>
      <th>age</th>
      <th>children</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>Kim</td>
      <td>24</td>
      <td>2</td>
    </tr>
    <tr>
      <th>1</th>
      <td>Lee</td>
      <td>27</td>
      <td>1</td>
    </tr>
    <tr>
      <th>2</th>
      <td>Park</td>
      <td>34</td>
      <td>3</td>
    </tr>
  </tbody>
</table>
</div>

In [127]:
result_df = pd.DataFrame({
    'name' : ['Kim', 'Lee', 'Park'],
    'age' : [24, 27, 34],
    'children' : [2, 1, 3]
})
result_df

name  age  children
0   Kim   24         2
1   Lee   27         1
2  Park   34         3

## 채점 수행

아래 코드는 채점 수행을 위한 코드입니다.

따라서 이를 수정했을시 **채점이 제대로 이루어지지 않습니다.**

**주의**: 채점 코드를 실행하기 전에 반드시 코드 파일을 한번 저장하시길 바랍니다.

In [128]:
import os

result_df.to_json('result.json')

os.system('elice_grade result.json cds_ds_exercise.ipynb')

1